test/games/strategy/util/UrlStreamsTest.java

```diff
@@ -0,0 +1,79 @@
+package games.strategy.util;
+
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class UrlStreamsTest {
+
+  private UrlStreams testObj;
+
+  private URL fakeUrl;
+
+  @Mock
+  private URLConnection mockUrlConnection;
+  @Mock
+  private InputStream mockInputStream;
+
+
+  @Before
+  public void setup() throws Exception {
+    // set up the test object with a function that will return a mocked url connection
+    testObj = new UrlStreams(url -> mockUrlConnection);
```

Comment:
instead of getting a live HTTP connection, we will return our mockUrlConnection object. We can then control what happens from there. 

Important to this change we will verify that caching is turned off, that the appropriate calls are made to the mock URL connection to turn caching off.